src/Tour.js

```diff
@@ -293,9 +311,37 @@ class Tour extends Component {
 
   gotoStep = n => {
     const { steps, getCurrentStep } = this.props
+    const startingStep = this.state.current;
+
     this.setState(prevState => {
       const nextStep = steps[n] ? n : prevState.current
 
+      if (nextStep !== startingStep && this.props.deterministic) {
+        const startingNode = document.querySelector(steps[startingStep].selector) || undefined
+
+        if (nextStep > startingStep) {
+          // the next step is somewhere in the future list of steps
+          this.attemptExecution(steps[startingStep].postAction, startingNode)
+
+          steps.slice(startingStep, nextStep).forEach((interimStep) => {
```

Comment:
This seems to be wrong - it should be `steps.slice(startingStep, nextStep + 1)`. Clicking on the dot of a step that has an preAction attached won't be executed without the +1.